Madhuri-TC0010-SearchProductCombOfCharAndInt

madhuri-210 opened the pull request.

Commit:
Added feature file and search module

Commit:
Added feature file and search module

sarangholey commented on Batch8CucumberSeleniumAPI/src/test/resources/features/ui/Search.feature:
```diff
@@ -23,4 +23,12 @@ Feature: Search functionality
\ No newline at end of file
+	  
+	@TC0010 
+	 Scenario:In Search Box If End User Give Combination Of (Symbol Characters And Number) As A Input Its Give An Error As NO Result(Given Input) Try Checking Your Spelling Or Use More General Term.
```
Try to make the scenario in lesser words

akashdktyagi commented on Batch8CucumberSeleniumAPI/src/test/resources/features/ui/Search.feature:
```diff
@@ -23,4 +23,12 @@ Feature: Search functionality
+	 Scenario:In Search Box If End User Give Combination Of (Symbol Characters And Number) As A Input Its Give An Error As NO Result(Given Input) Try Checking Your Spelling Or Use More General Term.
+		Given I have browser opened and url is navigated
+		When I search for product as "Del123"
+		Then Search result is displayed Try Checking Your Spelling Or Use More General Term. 
```
Try writing the error message in quotes, that way it is more readable. Smthng like:
Then Search result is displayed  as 'Try Checking Your Spelling Or Use More General Term'.

akashdktyagi commented on Batch8CucumberSeleniumAPI/src/test/resources/features/ui/Search.feature:
```diff
@@ -23,4 +23,12 @@ Feature: Search functionality
 		When I search for product as "Dell"
 		And I click on any product in the Search Result
 		Then I am able to see product description and detail in new tab
-	  
```
remove this extra line

Commit:
Once again added feature file n search module

akashdktyagi commented on Batch8CucumberSeleniumAPI/src/test/java/runner/RunTempUI.java:
```diff
@@ -8,9 +8,9 @@
+		features="classpath:features/ui/",
 		glue="stepdefs.ui",
-		tags="@TC0008",
+		tags="@TC0010",
```
This file should not be commited. Only commit files which are relevant to the task you are currently working on. Revert the changes.

akashdktyagi replied:
revert this file please.

akashdktyagi commented on Batch8CucumberSeleniumAPI/src/test/java/stepdefs/ui/SearchStepDefs.java:
```diff
@@ -117,6 +117,8 @@ public void i_am_able_to_see_product_description_and_detail_in_new_tab() {
 	}
 	
 	
+
```
remove these extra lines from here.

akashdktyagi commented on Batch8CucumberSeleniumAPI/src/test/resources/features/ui/Search.feature:
```diff
@@ -23,4 +23,12 @@ Feature: Search functionality
+	 Scenario:When user enter combination of character  and number in search test box it should say Try Checking Your Spelling.
+		Given I have browser opened and url is navigated
+		When I search for product as "Del123"
+		Then Search result should displayed Try Checking Your Spelling Or Use More General Term. 
```
`Scenario:` When user enter incorrect sequence of characters and number in search test box, application should validation error message 
		Given I have browser opened and url is navigated
		When I search for product as "Del123"
		Then I should get search result error message as  "Try Checking Your Spelling Or Use More General Term."

Commit:
Added CmnPageObject file

Commit:
In SearchPageObject file added ValidateInvalidateMessage and in
SearchStepdefs file added then(search_result_should
displayed_try_checking) method

Commit:
in SearchStepDefs and SearchPageObjects added previous method

Commit:
Added Search.feature file(TC0010),SearchPageObjects and SearchStepDefs
file(ValidateInvalidateMessage)

akashdktyagi commented on Batch8CucumberSeleniumAPI/src/test/java/stepdefs/ui/SearchStepDefs.java:
```diff
@@ -75,15 +75,19 @@ public void i_have_browser_opened_and_url_is_navigated() throws Exception {
+
+	@Then("Search result should displayed {string}")
+	public void search_result_should_displayed_Try_Checking_Your_Spelling_Or_Use_More_General_Term(String productName) {
+		testContextUI.getSearchPageObjects().ValidateInvalidateProductMessage();
```
Pass the argument to the method and validate that:
it should look like this:

`testContextUI.getSearchPageObjects().ValidateInvalidateProductMessage(productName);`

akashdktyagi commented on Batch8CucumberSeleniumAPI/src/test/java/pageobjects/SearchPageObjects.java:
```diff
@@ -70,6 +67,11 @@ public void ValidateProductList(String productName) {
-		
+	}
+	public void ValidateInvalidateProductMessage(){
+		String expected="Try Checking Your Spelling Or Use More General Term.";
```
Accept the argument, this method should look like this.
`public void ValidateInvalidateProductMessage(String errMessage){`
`String actual=getText(result_invalid_product);
		Assert.assertEquals(errMessage, actual);
		scn.write("Error message is displayed");`

Commit:
Once again SearchPageObjects and SearchStepDefs files added.
